src/System.Management.Automation/engine/Utils.cs

```diff
@@ -1399,66 +1399,17 @@ internal static Encoding GetEncoding(string path)
                 return ClrFacade.GetDefaultEncoding();
             }
 
-            byte[] initialBytes = new byte[100];
-            int bytesRead = 0;
-
             try
             {
-                using (FileStream stream = System.IO.File.OpenRead(path))
+                using (StreamReader sr = new StreamReader(path, true))
```

Comment:
I'm lost now, I thought we wanted to use the `StreamReader` to get the current file encoding, if the file already existed. Then keep that encoding. 

If I use the constructor with the Encoding, it will open the file with utf8NoBOM regardless of what the file's encoding is really. The sr.Encoding will always return utf8NoBOM, which means that we don't actually need to use the StreamReader in the first place.

I'm surely missing something.

Reply:
@Gimly 

> If I use the constructor with the Encoding, it will open the file with utf8NoBOM regardless of what the file's encoding is really

If we set the detection flag StreamReader will always call [DetectEncoding()](https://source.dot.net/#System.Private.CoreLib/StreamReader.cs,627) at first read operation. If it can not detect the file encoding it will use default. So we should set the default.

Reply:
You're right sorry, I misread StreamReader's documentation. I've pushed an update with the default